sdk/src/types/block/address/multi.rs

```diff
@@ -97,8 +98,9 @@ impl MultiAddress {
     pub fn new(addresses: impl IntoIterator<Item = WeightedAddress>, threshold: u16) -> Result<Self, Error> {
         let mut addresses = addresses.into_iter().collect::<Box<[_]>>();
 
-        addresses.sort_by(|a, b| a.address().cmp(b.address()));
+        addresses.sort_by(|a, b| a.address().pack_to_vec().cmp(&b.address().pack_to_vec()));
```

Comment:
I recognize that this is not likely to present a significant performance concern, but we have really no idea how many times this closure will be called. It would be better to generate a map (BTreeMap, probably) with the keys being the packed representation, then iterate through the map in order to create the sorted list.

Reply:
Fair enough

Reply:
agree ... and out of interest: can you maybe explain to me what this change corresponds to in the TIP?

Reply:
https://github.com/iotaledger/tips/blob/tip52/tips/TIP-0052/tip-0052.md#additional-syntactic-transaction-validation-rules

Reply:
Done